### PR TITLE
Add ELKS system call and C86 compiler helper libraries

### PR DESCRIPTION
**Pull request by ghaerr**
This adds two new files, lib/c86lib.asm which contains all helper routines the C86 compiler may emit, and lib/syscall.asm which contains wrappers for all ELKS system calls.

These files will allow you to eliminate clib.s/clib.o entirely and replace with lib/c86lib.o and lib/syscall.o, used when linking any ELKS executable. The lib/c86lib.o object file is only needed for certain 32-bit operations. 

I would recommend that .asm become the standard filename extension for NASM files. I didn't add these to be automatically built in the make, I figure you can do that after testing. After we get a working AR86, then these can be used to build the first ELKS libc.a which will contain these two and other files.

Note: these are generally untested but should work :) I also added \_errno to the syscall routine.

### CODE REVIEW

**Comment by ghaerr** [comment]
I just happened to notice some `retf` (rather than `ret`) instructions in c86lib.asm; these came from the BYU clib.s. I don't think we want RETF as we're not producing large model executables on ELKS (only using OWC large model to produce small model executables).

I have made a note of this. Lets no modify the RETF -> RET until I get C86 to actually produce calls to these functions, so I can see exactly what is happening.

Ok to commit, rather than wait. If needed I will fix later, and these aren't called unless 32-bit arithmetic is performed.

**Comment by rafael2k** [comment]
Very cool!! Thank @ghaerr !

**Comment by ghaerr** [comment]
More on the RETF in the C86 library: it seems that RETF is correct, as they're some funny business going on in the function prologue of each of those functions:
```
        pop    bx
        push   cs
        push   bx
```
As you can see, this pops IP then pushes CS:IP, so RETF is required. 

This isn't needed in our use, I'll take a further look at why/what might be the reason. I suspect it might have something to do with the BYU emulator doing something special when a RETF is interpreted. For instance, this could allow for an automatic interpreter stop after C86 compiler functions are executed, allowing for further debugging inspection.

In conclusion, we're all right for now, but the routines are slower than they need to be.
